### PR TITLE
refactor: clean usage of delimiters

### PR DESCRIPTION
**Pull request by pmiossec**
## Proposed changes

- No behavior change, just refactoring.

## Test methodology 

- Unit tests & manual

## Test environment(s) 

- GIT 
- Windows 

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by gerhardol — APPROVED** [review]
There are some occurrences of Split("string") instead of Split('c') that could be changed too

**Review by RussKie — APPROVED** [review]

File: GitExtUtils/Delimiters.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ internal static class Delimiters
         public static readonly char[] TabAndLineFeedAndCarriageReturn = { '\t', '\n', '\r' };
         public static readonly char[] NullAndLineFeed = { '\0', '\n' };
         public static readonly char[] LineFeedAndCarriageReturn = { '\n', '\r' };
+        public static readonly char[] LineFeedCarriageReturnAndNull = { '\n', '\r', '\0' };
```
Comment (mstv) [inline GitExtUtils/Delimiters.cs:21]: ❔
```suggestion
        public static readonly char[] LineFeedCarriageReturnAndZero = { '\n', '\r', '\0' };
```

File: Plugins/GitFlow/GitFlowForm.cs [hunk 2]
```diff
@@ -203,7 +203,7 @@ List<string> GetLocalBranches()
                 GitArgumentBuilder args = new("branch");
                 return _gitUiCommands.GitModule
                     .GitExecutable.GetOutput(args)
-                    .Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries).Select(e => e.Trim('*', ' ', '\n', '\r'))
+                    .Split('\n', StringSplitOptions.RemoveEmptyEntries).Select(e => e.Trim(Delimiters.GitOutput))
```
Comment (mstv) [inline Plugins/GitFlow/GitFlowForm.cs:206]:
```suggestion
                    .Split('\n', StringSplitOptions.RemoveEmptyEntries)
                    .Select(e => e.Trim(Delimiters.GitOutput))
```

**Review by mstv — COMMENTED** [review]
just nits

File: GitExtUtils/Delimiters.cs (hunk 1 above)
Comment (pmiossec) [reply GitExtUtils/Delimiters.cs]: I used the same naming than 2 lines above and I think we talk more about Null character https://en.wikipedia.org/wiki/Null_character so I prefer to use that.

**Review by mstv — APPROVED** [review]

**Pull request merged**